Paco-Server/ear/default/web/js/viz/vizService.js

```diff
@@ -9,91 +10,165 @@ pacoApp.factory('experimentsVizService', ['$http', 'experimentService', function
     return experiment;
   }
 
-  function getInputs(experiment) {
-    var responseDetails = [];
-    var inputs = experiment.groups[0].inputs;
-    inputs.forEach(function (input) {
-      if (input.responseType == "likert") {
-        responseDetails.push({
-          "name": input.name,
-          "responseType": input.responseType,
-          "text": input.text,
-          "leftsidelabel": input.leftSideLabel,
-          "rightsidelabel": input.rightSideLabel
-        });
-      } else if (input.responseType == "list") {
-        responseDetails.push({
-          "name": input.name,
-          "responseType": input.responseType,
-          "text": input.text,
-          "listChoices": input.listChoices
-        });
-      } else {
-        responseDetails.push({"name": input.name, "responseType": input.responseType, "text": input.text});
-      }
-    });
-    return responseDetails;
+  function getParticipants(experimentId) {
+
+    if (experimentId != undefined) {
+      var message = '{ "select":["who"], "query" : { "criteria" : "experiment_id = ?", "values" : [' + experimentId + ']},"group":"who"}';
+    }
+    var participants = httpPostBody(message);
+    return participants;
   }
 
-  function getEvents(id) {
+  function getDateRange(experimentId) {
 
-    var getEvents = $http({
+    var dateRange = [];
+    var format = 'MM/dd/yyyy';
+    if (experimentId != undefined) {
+      var endDateQuery = '{ "select":["response_time"], "query" : { "criteria" : "experiment_id = ?", "values" : [' + experimentId + ']},"order":"response_time desc","limit":"1"}';
+      var startDateQuery = '{ "select":["response_time"], "query" : { "criteria" : "experiment_id = ?", "values" : [' + experimentId + ']},"order":"response_time asc","limit":"1"}';
+    }
+    var startDate = httpPostBody(startDateQuery);
+    var endDate = httpPostBody(endDateQuery);
+
+    startDate.then(function (data) {
+
+      var format_startDate = $filter('date')(new Date(data.data.customResponse[0].response_time), format);
+      dateRange.push(format_startDate)
+    });
+    endDate.then(function (data) {
+      var format_endDate = $filter('date')(new Date(data.data.customResponse[0].response_time), format);
+      dateRange.push(format_endDate)
+    });
+    return dateRange;
+  }
+
+  function getEventsCounts(id) {
+    var eventsCount = [];
+    var getCount = $http({
       method: 'GET',
-      url: '/events?q=experimentId=' + id + '&json'
+      url: '/participantStats?experimentId=' + id + '&reportType=totalEventCounts&statv2=1'
     }).then(function successCallback(response) {
-      return response.data.events;
+      return response;
     }, function errorCallback(error) {
       return error;
     });
+    getCount.then(function (data) {
+      eventsCount.push(data.data[0].schedR, data.data[0].missedR, data.data[0].selfR);
+    });
+    return eventsCount;
+  }
+
+  function getEvents(experimentId, group, participants, startDate, startTime, endDate, endTime) {
```

Comment:
We need #1561 because that has the bower components to update angular - it is not obsolete. 
The overlap in the code is due to git merge from 'viz dashboard q1 q2' branch into 'data summary and UI enhancements' branch because I needed some of the code in the 'viz dashboard q1 q2' in order to complete task in the 'data summary and UI enhancements'. 

I will take note of your recommendations in the previous comments for the getEvents method and refactor it accordingly.